cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/Lookup.java

```diff
@@ -16,15 +16,43 @@
 
 package co.cask.cdap.etl.api;
 
+import co.cask.cdap.api.data.schema.Schema;
+
+import java.util.Collection;
 import java.util.Map;
 import java.util.Set;
+import javax.annotation.Nullable;
 
 /**
  * Exposes read-only lookup operations on datasets.
  *
- * @param <T> the type of object that will be returned for a lookup
+ * @param <T> the type of object that will be returned for a lookup with String as the key
+ * @param <R> the raw type of object that will be returned for lookups with byte[] as the key
```

Comment:
For KeyValueTable lookup, there is a method:
`String get(String key)`, where the key and value are both Strings.
There is also the method:
`byte[] get(String key)`, where the value type is different (both key and value are `byte[]`).